cli/cmd/uninstall.go

```diff
@@ -25,18 +27,34 @@ func newCmdUninstall() *cobra.Command {
 This command provides all Kubernetes namespace-scoped and cluster-scoped resources (e.g services, deployments, RBACs, etc.) necessary to uninstall Linkerd control plane.`,
 		Example: ` linkerd uninstall | kubectl delete -f -`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return uninstallRunE(cmd.Context())
+
+			k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+			if err != nil {
+				return err
+			}
+
+			if !force {
+				podsList, err := k8sAPI.CoreV1().Pods("").List(cmd.Context(), metav1.ListOptions{})
+				if err != nil {
+					return err
+				}
+				for _, pod := range podsList.Items {
+					// Skip core control-plane namespace
+					if pod.Namespace != controlPlaneNamespace && healthcheck.ContainsProxy(pod) {
+						return fmt.Errorf("Please uninject proxy containers before uninstalling the control-plane")
```

Comment:
We could collect the offending pods into a slice so that we can print the list of pods which need to be uninjected.